packages/dds/tree/src/simple-tree/proxies.ts

```diff
@@ -121,17 +122,63 @@ export function getOrCreateNodeProxy(flexNode: FlexTreeNode): TreeNode | TreeVal
 	}
 }
 
+/**
+ * Stores mappings from simple field keys ("view" keys) to their flex field counterparts ("stored" keys).
+ * Mappings are set via {@link cacheFlexKeyMapping}.
+ * Simple field keys can be translated via {@link getFlexKeyFromCache}.
+ */
+const simpleKeyToFlexKeyCache = new WeakMap<TreeNodeSchema, Map<string, FieldKey>>();
+
+/**
+ * Sets the view key to stored key mappings for the provided object field schemas in {@link simpleKeyToFlexKeyCache}.
+ */
+function cacheFlexKeyMapping(
+	nodeSchema: TreeNodeSchema,
+	fields: Record<string, ImplicitFieldSchema>,
+): void {
+	if (simpleKeyToFlexKeyCache.has(nodeSchema)) {
+		return;
+	}
+
+	const keyMap = new Map<string, FieldKey>();
+	for (const [viewKey, fieldSchema] of Object.entries(fields)) {
+		// Only specify mapping if the stored key differs from the view key.
+		// No entry in this map will indicate that the two keys are the same.
+		const storedKey = getStoredKey(viewKey, fieldSchema);
+		if (viewKey !== storedKey) {
+			keyMap.set(viewKey, brand(storedKey));
+		}
+	}
+
+	simpleKeyToFlexKeyCache.set(nodeSchema, keyMap);
+}
+
+/**
+ * Gets the flex domain key (i.e. the "stored key") associated with the provided view key in an object schema.
+ *
+ * If an explicit stored key was specified in the schema, it will be used. Otherwise,
+ * the stored key is the same as the view key.
+ */
+function getFlexKeyFromCache(viewKey: FieldKey, nodeSchema: TreeNodeSchema): FieldKey {
+	const keyMap = simpleKeyToFlexKeyCache.get(nodeSchema);
+	assert(keyMap !== undefined, "No key mappings were populated for the provided node schema.");
+	return keyMap.get(viewKey) ?? viewKey;
+}
+
 /**
  * @param allowAdditionalProperties - If true, setting of unexpected properties will be forwarded to the target object.
  * Otherwise setting of unexpected properties will error.
  * @param customTargetObject - Target object of the proxy.
  * If not provided `{}` is used for the target.
  */
-export function createObjectProxy<TSchema extends FlexObjectNodeSchema>(
-	schema: TSchema,
+export function createObjectProxy(
+	schema: TreeNodeSchema,
 	allowAdditionalProperties: boolean,
 	targetObject: object = {},
 ): TreeNode {
+	// Performance optimization: cache view key => stored key mapping.
+	cacheFlexKeyMapping(schema, schema.info as Record<string, ImplicitFieldSchema>);
```

Comment:
store the map here.

Ex:
const keyMap = .....;

Then you avoid all the cache lookups below.

Store all the keys in it, so users don't need ?? viewKey.

Then keyMap.has(key) can be used to implement `has` below and yuo can avoid the Object.keys call (and extra copy) in OwnKeys.

Reply:
also the map should output branded FieldKeys so users of them don't need to do branding

Reply:
Note that I'm just trying to clarify my previous suggestion here. I do this my suggested approach is a bit cleaner, but its not a big deal.

Reply:
I've updated the access pattern.

> Store all the keys in it, so users don't need ?? viewKey.

Since the common case is that there will be no explicit stored key, this seemed like a memory waste. But if we're confident the runtime bottleneck is the bigger risk, I can update.

Reply:
if we only do it once per schema, it should be fine. If we do it for every node, then that would be bad.

Reply:
I think what you did is also fine

Reply:
That's a good point about this only being per schema. Storing all of the keys makes this simpler. Updated.

Reply:
Actually, we still have to fall back to the input "view" keys in the case of symbols and additional properties, so we may as well preserve the space if the consumption patterns remain the same. Reverting the changes here.